pandas/tests/io/json/test_pandas.py

```diff
@@ -29,6 +29,20 @@
 )
 
 
+def test_literal_json_deprecation():
+    # PR 53409
+    expected = DataFrame([[1, 2], [1, 2]], columns=["a", "b"])
+    msg = (
+        "Passing literal json to 'read_json' is deprecated and "
+        "will be removed in a future version. To read from a "
+        "literal string, wrap it in a 'StringIO' object."
+    )
+
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        result = read_json('{"a": 1, "b": 2}\n{"b":2, "a" :1}\n', lines=True)
```

Comment:
Could you test that this also raises a warning with `lines=False`

Reply:
@mroeschke I added a check in the JsonReader class in `_json.py` to check for the presence of "\n" when lines == False. 

The check seems to be working correctly. However, execution continues after a warning is fired and the code ends up triggering a Value Error: `ValueError: Trailing data`. Seems like the ValueError is overriding the warning and the warning is never caught.

Is there any way to catch the Warning in this scenario?

Reply:
@mroeschke So just to make sure that I understand, are you asking that add additional tests, or make sure that execution stops once a warning is fired? Or both?